components/overflow-group/overflow-group.js

```diff
@@ -0,0 +1,468 @@
+			:host([dir="rtl"]) .d2l-overflow-group-container ::slotted(d2l-dropdown:not(.d2l-overflow-dropdown)),
+			:host([dir="rtl"]) .d2l-overflow-group-container ::slotted(d2l-dropdown-button),
+			:host([dir="rtl"]) .d2l-overflow-group-container ::slotted(d2l-dropdown-button-subtle),
+			:host([dir="rtl"]) .d2l-overflow-group-container ::slotted(d2l-dropdown-more),
+			:host([dir="rtl"]) .d2l-overflow-group-container ::slotted(d2l-dropdown-context-menu) {
+				margin-left: 0.6rem;
+				margin-right: 0;
+			}
+			:host([opener-style="subtle"]) .d2l-overflow-group-container ::slotted(d2l-button-subtle) {
+				margin-right: 0.2rem;
+			}
+			:host([opener-style="subtle"][dir="rtl"]) .d2l-overflow-group-container ::slotted(d2l-button-subtle) {
+				margin-left: 0.2rem;
+				margin-right: 0;
+			}
+			.d2l-overflow-group-container ::slotted([data-is-chomped]) {
+				display: none !important;
+			}`
+		];
+	}
+	constructor() {
+		super();
+		this._handleResize = this._handleResize.bind(this);
+
+		this._throttledResize = throttle(this._handleResize, 15);
+
+		this._overflowHidden = false;
+		this.autoShow = false;
+		this.maxToShow = -1;
+		this.minToShow = 1;
+		this.openerStyle = OPENER_STYLE.DEFAULT;
+		this.openerType = OPENER_TYPE.DEFAULT;
+		this._mini = this.openerType === OPENER_TYPE.ICON;
+
+		this._slotItems = [];
+	}
+
+	async firstUpdated() {
+		super.firstUpdated();
+
+		// selected elements
+		this._buttonSlot = this.shadowRoot.querySelector('slot');
+
+		this._container = this.shadowRoot.querySelector('.d2l-overflow-group-container');
+
+		this._getItems();
+
+		if (this.autoShow) {
+			this._autoDetectBoundaries(this._slotItems);
+		}
+
+		// this._availableWidth = this._container.clientWidth;
+
+		this.resizeObserver = new ResizeObserver(this._throttledResize);
+		this.resizeObserver.observe(this._container);
+	}
+	render() {
+		const overflowMenu = this._getOverflowMenu();
+
+		this._slotItems.forEach((element, index) => {
+			if (!this._overflowMenuHidden && index >= this._chompIndex) {
+				element.setAttribute('data-is-chomped', '');
+			} else {
+				element.removeAttribute('data-is-chomped');
+			}
+		});
+
+		return html`
+			<div class="d2l-overflow-group-container">
+				<slot @slotchange="${this._handleSlotChange}"></slot>
+				${overflowMenu}
+			</div>
+		`;
+	}
+	update(changedProperties) {
+		super.update(changedProperties);
+		if (changedProperties.get('autoShow')) {
+			this._getItemLayouts(this._slotItems);
+			this._autoDetectBoundaries(this._itemLayouts);
+		}
+
+		if (changedProperties.get('minToShow')
+			|| changedProperties.get('maxToShow')) {
+			this._chomp();
+		}
+
+		// Slight hack to get the overflow menu width the first time it renders
+		if (!this._overflowMenuWidth) {
+			// this action needs to be deferred until first render of our overflow button
+			requestAnimationFrame(() => {
+				this._chomp();
+			});
+		}
+	}
+	_autoDetectBoundaries(items) {
```

Comment:
Not for this PR: I'd still like to see when we port this over to the LMS if we can remove the need for it altogether. Seeing as it's just used once and maybe unnecessarily, it's an extra iteration over all the elements that we can maybe avoid.